dbt/adapters/athena/impl.py

```diff
@@ -240,6 +242,9 @@ def is_work_group_output_location_enforced(self) -> bool:
         conn = self.connections.get_thread_connection()
         creds = conn.credentials
 
+        if creds.work_group_enforced is not None:
```

Comment:
If you use a default to False, you can avoid to check for None, and simply return True esplicitely.
Pretty much we introduce a new param with a default value of False, and we check only when is set to True.

Reply:
I am not sure if you mean this snippet to be desired:

```
    @available
    def is_work_group_output_location_enforced(self) -> bool:
        conn = self.connections.get_thread_connection()
        creds = conn.credentials

        if not creds.work_group_enforced:
            return True

        if creds.work_group:
            work_group = self._get_work_group(creds.work_group)
            output_location = (

           ...
```

If yes, I don't understand why returning True in the first condition is better than returning False instead. The same doubt I would have if you proposed returning False instead of True. I see this value is later used to execute CTAS. Is there something I missed?

Reply:
I got confused myself by this implementation.

I understand the general idea, you want to skip the workgroup call via a parameter.

Now let's have a look at is_work_group_output_location_enforced function before your implementation:
* if an output location is set and if workgroup enforced -> the function returns True
* if none of the above is valid by default returns False

Now, with your parameter you want to actually skip the call on the worgroup simulating that the workgroup is enforced or not.

I suggest to use a parameter called `skip_workgroup_check`-> it can be True or False.

and the implementation can look like this:


```
if skip_workgroup_check is True:
   # we assume that the user do not care about using workgroup location, and 
   return False
else: 
  # current implementation
```

by default skip_workgroup_check MUST be False, to allow backwards compatibilty.

Doing so, you achieve what you want simply setting skip_workgroup_check to True, that return False.

Reply:
I changed the implementation to fit the meaning of introduced `skip_workgroup_check` parameter.